pkg/config/config.go

```diff
@@ -344,6 +344,9 @@ type loaderFunc func(path string, fileType string, expandArgs bool, target *Conf
 // load allows for tests to inject a function for retrieving the config file that
 // doesn't require having a literal file on disk.
 func load(fs *flag.FlagSet, args []string, loader loaderFunc) (*Config, error) {
+	// Set up the default logger parameters.
+	DefaultConfig.Server.LogLevel.Set("info")
+	DefaultConfig.Server.LogFormat.Set("json")
```

Comment:
Hmm, this might be a little unreliable and easy to accidentally remove during refactoring. (Also, the default log format should be logfmt, not json :)

It might be better for us to do a bit of a hack in `pkg/server.DefaultConfig` to make sure the defaults are set properly, something like:

```go
var DefaultConfig = Config{
  LogLevel: func() logging.Level {
     var l logging.Level
     // Pass a fake flag set to set the defaults
     l.RegisterFlags(flag.NewFlagSet("", flag.PanicOnError))
     return l
  }(),
  LogFormat: func() logging.Format {
     var l logging.Format
     l.RegisterFlags(flag.NewFlagSet("", flag.PanicOnError))
     return l
  }(),

  // ...
}
```

also, can you add a unit test for this and a changelog entry?

Reply:
Yeah, unfortunately this change trips up the linter for not checking the error. I was just trying to find a way to set it up directly in the DefaultConfig, but it would still be kinda ugly.

Reply:
We could probably make our own structs that we can default more easily, though that would be slightly more work for you